src/classes/devices/ble.types.ts

```diff
@@ -0,0 +1,30 @@
+export type UUID = string;
+
+export interface LimitedPeripheralData {
+  name: string;
+  id: string;
+  advertising: number[];
+  rssi: number;
+}
+
+export interface PeripheralData extends LimitedPeripheralData {
+  services?: UUID[];
+  characteristics: Characteristic[];
+}
+
+export interface Characteristic {
+  service: UUID;
+  chracterstic: UUID;
+  propertis: Property[];
+  descriptors?: Descriptor[];
+}
+
+export enum Property {
+  READ = 'Read',
```

Comment:
After enums are always a bit strange in typescript.
Wouldn't it be better to write:
READ = 'READ' 
WRITE = 'WRITE
NOTIFY = 'NOTIFY

So we don't come in any upper/lowercase issues

Reply:
Those are actual constants from BLE plugin, I cannot rename the values. https://github.com/don/cordova-plugin-ble-central#peripheral-data